.github/workflows/e2e-nvidia-t4-x1.yml

```diff
@@ -30,9 +30,34 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
   cancel-in-progress: true
 
+env:
+  LC_ALL: en_US.UTF-8
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  contents: read
+
 jobs:
+  e2e-ready:
+    permissions:
+      checks: read
+    uses: ./.github/workflows/status-checks.yml
+    with:
+      job_names: >- # Space-separated job names to wait on for status checks
+        DCO
+        actionlint
+        markdown-lint
+        shellcheck
+        spellcheck
+        lint-workflow-complete
+      delay: 5 # wait while jobs get started
```

Comment:
What happens without the delay?  I assume the logic is that it fails immediately if they don't exist?

Reply:
The delay allows the jobs queue to build up before deciding if the jobs are complete (or not present because they are not triggered). With a zero delay, it is possible this job could start first, see only a subset of the actual jobs and then exit with success since the named jobs are not yet present in the jobs queue. So I picked 5 seconds but we can pick another delay value if desired.

Reply:
Here is a current example where the workflow tests are kicking off for a first time user and the other jobs haven't:  https://github.com/instructlab/instructlab/pull/2242

I was mostly curious what would happen in that case.

Reply:
The wait-for-status-checks action does not require that a job has run. It only requires that if [check-runs](https://docs.github.com/en/rest/checks/runs?apiVersion=2022-11-28#list-check-runs-for-a-git-reference) reports the job, the job is skipped or successful. 

In the mentioned PR, check-runs shows:

```sh
➜  ~ curl -s -L -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" https://api.github.com/repos/instructlab/instructlab/commits/3c951f5a62a402523fb38ee0195e933a86b2003e/check-runs  | jq '.check_runs[] | {(.name): .status}'
{
  "e2e-workflow-complete": "completed"
}
{
  "Stop external EC2 runner": "completed"
}
{
  "E2E Test": "completed"
}
{
  "Summary": "completed"
}
{
  "Start external EC2 runner": "completed"
}
{
  "DCO": "completed"
}
```

so the wait-for-status-checks action would end with success.

I will make another change to split the e2e workflow into 2 parts: one pull_request, the other workflow_run. The first will be subject to approval for first time contributors and the second will have access to needed secrets. We used this technique  in the taxonomy repository for a workflow having similar concerns.